repos/os/src/drivers/platform/spec/x86/pci_config_access.h

```diff
@@ -81,7 +60,17 @@ namespace Platform {
 
 		public:
 
-			Config_access(Genode::Env &env) : _env(env) { }
+			Config_access(Env &env) : _env(env) {
+				Attached_rom_dataspace info(_env, "acpi");
+				Xml_node xml(info.local_addr<char>(), info.size());
+				for (unsigned i = 0; i < xml.num_sub_nodes(); i++) {
+					Xml_node node = xml.sub_node(i);
+
+					if (node.has_type("bdf")) {
+						node.attribute("base").value(&_base);
```

Comment:
`Xml_node` methods may throw, for example, `Nonexistent_attribute`, which should be catched. For convenience, you may use `Xml_node::attribute_value`. There's also `Xml::for_each_sub_node()` which seems to fit your use case.

Reply:
Let me add that the `Attached_rom_dataspace` has an `xml()` method, which alleviates the need to manually construct a `Xml_node`.

Reply:
Thanks for the input, I will simplify the code accordingly.

Reply:
I think this code is not correct. The "bdf" element may appear more than once. This "bdf" node correlates to the mcfg table, which may report several mmconf regions for some part of the pci bdf space. So, you will have to add here a Object which stores this base address and the size, and the pci range this mmconf is fore.

Reply:
I meant a list of such mmconf object ranges and you would have to adjust this "_base + _dev_base(bdf)" calculation by going effectively through this list.

Reply:
As discussed on IRC, I will adjust the code as follows:
- Move handling of PCI config space mapping to early platform drv boot code (`Platform::Root)`
  - Throw error if multiple bdf nodes exist
  - This is a simplification since we have not yet encountered systems with multiple PCI segment groups
  - Throw error if no node for segment 0:0.0 exists
  - Map complete PCI segment group region for 0:0.0 early in the platform drv boot code
- Pass on PCI config space mapping reference to session, device, config access